src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -225,20 +231,25 @@ public Response addGzip(InputStream gzip)
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
 			return builder.addError(e.getMessage()).build();
 		}
-		return (add(gzipInputStream));
+		return (add(httpHeaders, gzipInputStream));
 	}
 
 	@POST
 	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
 	@Path("/datapoints")
-	public Response add(InputStream json)
+	public Response add(@Context HttpHeaders httpHeaders, InputStream json)
 	{
-		try
+
+
+		Span span = createSpan("datapoints_insert", httpHeaders);
+
+		try (Scope scope = tracer.scopeManager().activate(span, false))
 		{
 			DataPointsParser parser = new DataPointsParser(datastore, new InputStreamReader(json, "UTF-8"),
 					gson, m_kairosDataPointFactory);
 			ValidationErrors validationErrors = parser.parse();
 
+			span.setTag("datapoint_count", parser.getDataPointCount());
```

Comment:
I believe it does make sense as I would like to know how many datapoints are actually fetched/written in kairosdb. Adding to the log is also fine. However, I think having them as tag makes it more searchable.

Reply:
How will you search the count?

Reply:
If we have better search features in future do we change the instrumentation again?

Reply:
Actually this argument is in favor of instrumentation assuming least capabilities. We need to work on basis of how can we get the best value of our instrumentation assuming lowest features on the backend side. 
It also helps better tagging/logging practice, since having those advanced regex based search for instance, could lead to jamming all values in one tag and ending up with a worse tracing + getting locked into single backend.